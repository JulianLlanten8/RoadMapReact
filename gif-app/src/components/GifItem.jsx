import propTypes from "prop-types";
export const GifItem = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__fadeIn">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

GifItem.propTypes = {
  title: propTypes.string.isRequired,
  url: propTypes.string.isRequired,
};
