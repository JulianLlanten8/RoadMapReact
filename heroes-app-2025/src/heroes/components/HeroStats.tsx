import { Badge } from "@/components/ui/badge";
import { Heart, Trophy, Users, Zap } from "lucide-react";
import { HeroStatCars } from "./HeroStatCars";

export const HeroStats = () => {
  return (
    <div className="grid grid-cols-2 md:grid-cols-4 gap-4 mb-8">
      <HeroStatCars
        title="Total Characters"
        icon={<Users className="h-4 w-4 text-muted-foreground" />}
      >
        <div className="text-2xl font-bold"></div>
        <div className="flex gap-1 mt-2">
          <Badge variant="secondary" className="text-xs">
            Heroes
          </Badge>
          <Badge variant="destructive" className="text-xs">
            Villains
          </Badge>
        </div>
      </HeroStatCars>

      <HeroStatCars
        title="Favoritos"
        icon={<Heart className="h-4 w-4 text-muted-foreground" />}
      >
        <div className="text-2xl font-bold text-red-600">3</div>
        <p className="text-xs text-muted-foreground">% of total</p>
      </HeroStatCars>

      <HeroStatCars
        title="Strongest"
        icon={<Zap className="h-4 w-4 text-muted-foreground" />}
      >
        <div className="text-lg font-bold">Super Man</div>
        <p className="text-xs text-muted-foreground">Strength: 10/10</p>
      </HeroStatCars>

      <HeroStatCars
        title="Inteligente"
        icon={<Trophy className="h-4 w-4 text-muted-foreground" />}
      >
        <div className="text-lg font-bold">Batman</div>
        <p className="text-xs text-muted-foreground">Intelligence: 10/10</p>
      </HeroStatCars>

      {/* 

       */}
    </div>
  );
};
