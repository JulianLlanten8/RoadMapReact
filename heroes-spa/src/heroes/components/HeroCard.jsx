import PropTypes from "prop-types";
import { useNavigate } from "react-router-dom";
export const HeroCard = ({
  id,
  superhero,
  publisher,
  alter_ego,
  first_appearance,
  characters,
}) => {
  const heroImg = `/assets/heroes/heroes/${id}.jpg`;
  const navigate = useNavigate();
  return (
    <div className="col p-3">
      <div
        role="button"
        className="card bg-dark hover-shadow h-100 zoom cursor-pointer"
        onClick={() => {
          //ir a la ruta /hero/id
          navigate(`/hero/${id}`);
        }}
      >
        <img
          src={heroImg}
          className="card-img-top"
          style={{ minWidth: "200px" }}
          alt={superhero}
        />
        <div className="card-body">
          <h5 className="card-title text-info">{superhero}</h5>
          <p className="card-text text-light">{alter_ego}</p>
          {alter_ego !== characters && (
            <p className="card-text text-white-50">{characters}</p>
          )}
        </div>
        <div className="card-footer">
          <small className="text-white-50">{first_appearance}</small>
        </div>
      </div>
    </div>
  );
};

HeroCard.propTypes = {
  id: PropTypes.string.isRequired,
  superhero: PropTypes.string.isRequired,
  publisher: PropTypes.string.isRequired,
  alter_ego: PropTypes.string.isRequired,
  first_appearance: PropTypes.string.isRequired,
  characters: PropTypes.string.isRequired,
};
