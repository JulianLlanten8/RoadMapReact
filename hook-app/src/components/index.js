export * from "./LoandingQuote";
export * from "./Quote";
export * from "./TodoAdd";
export * from "./TodoList";
export * from "./TodoListItem";
export * from "./Navbar";
