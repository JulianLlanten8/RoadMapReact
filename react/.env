REACT_APP_API_KEY_GIPHY = "n7X50L3fGzjK3N84gA7yPI6PLxzrIhS0"
