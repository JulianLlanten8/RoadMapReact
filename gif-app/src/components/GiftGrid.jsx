import propTypes from "prop-types";

import { GifItem } from "./GifItem";
import { useFetchGifs } from "../hooks/useFetchGifs";

export const GiftGrid = ({ category }) => {
  const { images, isloading } = useFetchGifs(category);
  console.log({ images, isloading });

  return (
    <>
      <h3>{category}</h3>
      {isloading && <h2>Cargando...</h2>}
      <main className="card-grid">
        {images.map((image) => (
          <GifItem key={image.id} {...image} />
        ))}
      </main>
    </>
  );
};

GiftGrid.propTypes = {
  category: propTypes.string.isRequired,
};
