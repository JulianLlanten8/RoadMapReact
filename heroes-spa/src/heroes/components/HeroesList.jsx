import { getHeroByPublicer } from "../helpers";
import { HeroCard } from "./HeroCard";

export const HeroesList = ({ publicer }) => {
  const heroes = getHeroByPublicer(publicer);
  return (
    <div className="row rows-cols-1 rows-cols-md-3 g-3">
      {heroes.map((hero) => (
        <HeroCard key={hero.id} {...hero} />
        
      ))}
    </div>
  );
};
