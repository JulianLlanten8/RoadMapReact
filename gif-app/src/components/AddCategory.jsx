import { useState } from "react";
import propTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");
  const onInputChange = ({ target }) => {
    setInputValue(target.value);
  };
  const onSubmit = (event) => {
    event.preventDefault();
    const newInputValue = inputValue.trim();
    if (newInputValue.length < 2) return;
    // setCategories((categories) => [newInputValue, ...categories]);
    onNewCategory(newInputValue);
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit} role="form">
      <input
        type="text"
        name="AddCategory"
        id="newCategory"
        placeholder="Buscar gifs..."
        value={inputValue}
        onChange={onInputChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  // setCategories: propTypes.func.isRequired,
  onNewCategory: propTypes.func.isRequired,
};
