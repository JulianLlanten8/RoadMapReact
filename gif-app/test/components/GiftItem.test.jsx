/* eslint-disable no-undef */
import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("GiftItem", () => {
  const title = "pokemon";
  const url = "https://localhost/pokemon.jpg";

  test("debe hacer match con el snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);

    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar la imagen con el url y el alt indicado", () => {
    render(<GifItem title={title} url={url} />);
    /* screen.debug(screen.getByRole("img"));
    expect(screen.getByRole("img").src).toBe(url); */

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test("Debe de mostrar el titulo de lÃ¡ imagen", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
