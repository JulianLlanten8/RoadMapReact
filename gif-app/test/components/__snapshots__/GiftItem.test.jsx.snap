// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GiftItem debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card animate__animated animate__fadeIn"
  >
    <img
      alt="pokemon"
      src="https://localhost/pokemon.jpg"
    />
    <p>
      pokemon
    </p>
  </div>
</div>
`;
